README.md

```diff
@@ -1,21 +1,131 @@
-# Pesto Assignments
+# How Browsers work
 
-Welcome to Pesto assignments.
+## What is the main functionality of the browser?
 
+The primary function of a browser is to present the web resource you choose, by requesting it from the server and displaying it in the browser window. The resource is usually an HTML document, but may also be a PDF, image, or some other type of content. The location of the resource is specified by the user using a URI (Uniform Resource Identifier).
 
-## How to solve assignments
+The way the browser interprets and displays HTML files is specified in the HTML and CSS specifications. These specifications are maintained by the W3C (World Wide Web Consortium) organization, which is the standards organization for the web.
 
-Weekly Assignments will be shared by your mentor or instructor.
+## High-Level Components of a browser?
```

Comment:
Before jumping into High level components, You could have mentioned about DNS Resolution , TCP/IP etc ...